.github/workflows/build.yml

```diff
@@ -308,3 +310,194 @@ jobs:
+            });
+          }
+
+          // Call the GitHub API to get a release by tag
+          let release = null;
+          try {
+            release = await github.rest.repos.getReleaseByTag({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              tag: 'nightly',
+            });
+            console.log(`Found release ${release.data.tag_name} ${release.data.draft} ${release.data.prerelease}`);
+          } catch (err) {
+            console.log(`Release ${tag} not found`);
+
+            // If the release doesn't exist, create it
+            release = await github.rest.repos.createRelease({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              tag_name: 'nightly',
+              name: 'nightly',
+              body: 'Nightly release crafted with ♥️ somewhere on 🌎',
+              draft: false,
+              prerelease: true,
+            });
+
+            console.log(`Created release ${release.data.tag_name} ${release.data.draft} ${release.data.prerelease} ${doIProceed}`);
+          }
+          console.log(`Release does exist with tag ${release.data.tag_name} [${release.data.draft} ${release.data.prerelease}]`);
+
+          // At this stage both tag & release exist
+
+          // Update nightly tag
+          await github.rest.git.updateRef({
+            owner: context.repo.owner,
+            repo: context.repo.repo,
+            ref: 'tags/nightly',
+            sha: ref_head.data.object.sha,
+          });
+          console.log(`Updated tag with sha ${ref_nightly.data.object.sha}`);
+
+          // Update the release
+          await github.rest.repos.updateRelease({
+            owner: context.repo.owner,
+            repo: context.repo.repo,
+            release_id: release.data.id,
+            tag_name: 'nightly',
+            name: 'nightly',
+            body: 'Nightly release crafted with ♥️ somewhere on 🌎',
+            draft: false,
+            prerelease: true,
+          });
+          console.log(`Updated ${release.data.tag_name} nightly release  ${release.data.draft} ${release.data.prerelease}`);
+
+          // Get all tags and keep the oldest one starting by v
+          let oldest_tag = { name: 'v0.0.0' };
+
+          try {
+            const tags = await github.rest.repos.listTags({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+            });
+
+            // Keep latest tag
+            for (const tag of tags.data) {
+              if (tag.name.startsWith('v')) {
+                // if (tag.name > oldest_tag.name) {
+                if (tag.name.localeCompare(oldest_tag.name, undefined, { numeric: true, sensitivity: 'base'}) > 0) {
+                  oldest_tag = tag;
+                }
+              }
+            }
+          } catch (err) {
+            console.log(err);
+            throw new Error('Failed to get tags');
+          }
```

Comment:
There are several occurrences of
```js
try {
  ...
} catch(err) {
  console.log(err)
  throw new Error(...);
}
```
This is useless, remove the try/catch, the error message will already get displayed in log.

Reply:
done